packages/cli/src/core/interfaces/Extension.ts

```diff
@@ -0,0 +1,24 @@
+import { BytesLike } from "ethers";
+
+export interface ExtensionMetadata {
+    name: string;
+    metadataURI: string;
+    implementation: string;
+}
+
+export interface ExtensionFunction {
+    functionSelector: BytesLike;
+    functionSignature: string;
+}
+
+export interface Extension {
+    metadata: ExtensionMetadata;
+    functions: ExtensionFunction[];
+}
+
+export interface ExtensionDeployArgs {
+    name: string;
+    amount: number;
+    salt: string;
+    bytecode: string;
```

Comment:
there's a little bit of duplicated code with what's in the SDK (in yash's marketplace v3 branch) should add a TODO here to consolidate